### DISCUSSION TITLE
How to organize control flow with errors on useRouteLoaderData?

### DISCUSSION THREAD

**Discussion by WesleyYue** [remix-run/remix#5400] [Q&A]
I have a parent component and a child component, each with their own loaders. The child component requires data from the parent component, so I use useRouteLoaderData pointing to the parent. This parent loader is loading some data that is not critical to the child component. What is the correct Remix paradigm to handle errors in the parent loader, while being called from the child component? If this was a regular API endpoint, I would just check if the fetch was successful and handle it appropriately, but with the useRouteLoaderData, I'm not sure how I am supposed to detect and handle this error in the child component.

**Comment by kiliman** [comment]
`useRouteLoaderData` and, by extension, `useLoaderData` and `useMatches` do not call the loader. The loaders have already been fetched, and the data is stored in the Remix context. These hooks simply access the data that is in context.

If your parent loader had an error, it would render the ErrorBoundary and not your child component.

**Reply by WesleyYue** [reply]
Right, I understand, what I would like to do is not have it render the ErrorBoundary so that I can handle it in the child component. ~~For additional context, the parent route component is not rendered in the child route, it just has the loader.~~

EDIT: I was mistaken, the route I want to load from is actually a sibling route component, not a parent component

**Reply by WesleyYue** [reply]
For example, my desired interface would be something like:

```
const [data, err] = useRouteLoaderData(...)
return (<><RestOfMyComponent/>{ !err && <PartOfMyComponentThatNeedsSiblingLoaderData /> }<>)
```

But maybe there's a different way to solve this in a Remix way that I'm not understanding?

**Reply by WesleyYue** [reply]
Or a potentially even more common scenario that's simpler:

Let's say you have a component that needs 2 pieces of data from its own loader. In the loader, you fetch for piece1 and piece2. Let's say piece2 fails for some unforeseen reason, but we still want to render the component with what we have from piece1 since that's the important bit. What is the correct way to implement this?

I guess you MUST split the component into further nested components and use a catch boundary on the piece2 component that is factored out?
